csFastFloat/FastDoubleParser.cs

```diff
@@ -420,40 +448,52 @@ unsafe static internal AdjustedMantissa ParseLongMantissa(byte* first, byte* las
     }
 
 
-    unsafe static internal double HandleInvalidInput(char* first, char* last)
+    unsafe static internal double HandleInvalidInput(char* first, char* last, out long characters_consumed)
     {
       if (last - first >= 3)
       {
         if (Utils.strncasecmp(first, "nan", 3))
         {
+          characters_consumed = 3;
```

Comment:
As side-note:
```suggestion
          characters_consumed = "nan".Length;
```
The JIT will emit the same code, as it can handle const strings.

But I'll keep the code as is.